src/typechecker.erl

```diff
@@ -854,6 +854,9 @@ int_range_to_types({neg_inf, I}) when I > 0 ->
     [type(neg_integer),
      {type, erl_anno:new(0), range, [{integer, erl_anno:new(0), 0}
                                     ,{integer, erl_anno:new(0), I}]}];
+int_range_to_types({neg_inf, I}) -> % I < -1
```

Comment:
Why not add `when I < -1` as an actual guard, instead of a comment?

Reply:
Code is better than comments :-)
I somehow was thinking that it'd be easier for the exhaustiveness checker to see that the cases were exhaustive this way. But I was clearly thinking backwards, we shouldn't necessarily make things easy for the exhaustiveness checker, we should make it understand the code we write.